core/native_test.go

```diff
@@ -0,0 +1,344 @@
+package core
+
+import (
+	"github.com/QuarkChain/goquarkchain/account"
+	"github.com/QuarkChain/goquarkchain/core/types"
+	"github.com/QuarkChain/goquarkchain/serialize"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+
+	//"github.com/QuarkChain/goquarkchain/account"
+	"github.com/QuarkChain/goquarkchain/common"
+	//"github.com/QuarkChain/goquarkchain/core"
+	//"math/big"
+	"testing"
+)
+
+var (
+	testShardCoinbaseAmount2 = new(big.Int).Mul(new(big.Int).SetUint64(200), jiaozi)
+)
+
+func TestNativeTokenTransfer(t *testing.T) {
+	QETH := common.TokenIDEncode("QETH")
+	QKC := common.TokenIDEncode("QKC")
+	id1, _ := account.CreatRandomIdentity()
+	acc1 := account.CreatAddressFromIdentity(id1, 0)
+	acc2 := account.CreatEmptyAddress(0)
+	acc3 := account.CreatEmptyAddress(0)
+	testGenesisMinorTokenBalance["QKC"] = big.NewInt(10000000)
+	testGenesisMinorTokenBalance["QETH"] = big.NewInt(99999)
+	env := getTestEnv(&acc1, nil, nil, nil, nil, nil)
+	shardState := createDefaultShardState(env, nil, nil, nil, nil)
+	val := big.NewInt(12345)
+	gas := uint64(21000)
+	gasPrice := uint64(1)
+	tx1 := createTransferTransaction(shardState, id1.GetKey().Bytes(), acc1, acc2, val, &gas, &gasPrice, nil, nil, nil, &QETH)
+	error := shardState.AddTx(tx1)
+	if error != nil {
+		t.Errorf("addTx error: %v", error)
+	}
+	b1, _ := shardState.CreateBlockToMine(nil, &acc3, nil, nil, nil)
+	assert.Equal(t, len(b1.Transactions()), 1)
+	shardState.FinalizeAndAddBlock(b1)
+	assert.Equal(t, shardState.CurrentHeader(), b1.Header())
+	tokenBalance, _ := shardState.GetBalance(id1.GetRecipient(), nil)
+	tokenBalance2, _ := shardState.GetBalance(acc2.Recipient, nil)
+	tokenBalance3, _ := shardState.GetBalance(acc2.Recipient, nil)
+	qkcb := tokenBalance.GetTokenBalance(QKC)
+	assert.Equal(t, qkcb, big.NewInt(10000000-21000))
+	assert.Equal(t, tokenBalance.GetTokenBalance(QETH), big.NewInt(99999-12345))
+	assert.Equal(t, tokenBalance2.GetTokenBalance(QETH), big.NewInt(12345))
+	reward := new(big.Int).Add(testShardCoinbaseAmount, big.NewInt(21000)).Uint64()
+	assert.Equal(t, tokenBalance3.GetTokenBalance(QKC), afterTax(reward, shardState))
+	tTxList, _, err := shardState.GetTransactionByAddress(acc1, nil, nil, 0)
```

Comment:
tTxlist is nil  that not true

Reply:
https://github.com/QuarkChain/pyquarkchain/blob/08aa932343780279254cdd30dd665a090f931d46/quarkchain/cluster/tests/test_native_token.py#L81